provisioner/registryv1/controllers/bundleinstance_controller.go

```diff
@@ -93,21 +96,19 @@ func (r *BundleInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 
 	bi := &olmv1alpha1.BundleInstance{}
 	if err := r.Get(ctx, req.NamespacedName, bi); err != nil {
-		if apierrors.IsNotFound(err) {
-			return ctrl.Result{}, nil
-		}
-		return ctrl.Result{}, err
+		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 	defer func() {
 		bi := bi.DeepCopy()
 		bi.ObjectMeta.ManagedFields = nil
-		if err := r.Status().Patch(ctx, bi, client.Apply, client.FieldOwner("core.rukpak.io/registry+v1")); err != nil {
+		if err := r.Status().Patch(ctx, bi, client.Apply, client.FieldOwner(registryV1ProvisionerID)); err != nil {
 			l.Error(err, "failed to patch status")
 		}
 	}()
 
 	b := &olmv1alpha1.Bundle{}
 	if err := r.Get(ctx, types.NamespacedName{Name: bi.Spec.BundleName}, b); err != nil {
+		bi.Status.InstalledBundleName = ""
```

Comment:
Should we reset the `Status.InstalledBundleName` only when the error is NotFound? If we reset the field on any error, then the following could occur:
* Bundle is on cluster and installed
* BI controller reconciles and makes progress
* Transient error getting the Bundle occurs early in the control loop
* BI `Status.InstalledBundleName` gets reset 

I guess the next time the bundle is found the status is updated, but I'm wondering if instead it makes more sense to only reset the field if the bundle is not found

Reply:
Yep, I think that makes sense. I updated in the last force push to only reset when apierrors.IsNotFound(...) is true.